vm/actor/src/builtin/paych/mod.rs

```diff
@@ -128,7 +128,13 @@ impl Actor {
         }
 
         if !sv.secret_pre_image.is_empty() {
-            let hashed_secret: &[u8] = &rt.syscalls().hash_blake2b(&params.secret)?;
+            let hashed_secret: &[u8] =
+                &rt.syscalls().hash_blake2b(&params.secret).map_err(|e| {
+                    ActorError::new(
+                        ExitCode::ErrPlaceholder,
+                        format!("hashing input error: {}", e),
+                    )
+                })?;
```

Comment:
This is an example when you'd have to downcast the error returned. Don't assume a placeholder exit code, we will just run into inconsistencies